src/model/wallet.ts

```diff
@@ -342,49 +478,63 @@ export const getWallet = walletSeed => {
 };
 
 export const createWallet = async (
-  seed = null,
-  color = null,
-  name = null,
-  overwrite = false
-) => {
+  seed: null | EthereumSeed = null,
+  color: null | number = null,
+  name: null | string = null,
+  overwrite: boolean = false
+): Promise<null | EthereumWallet> => {
   const isImported = !!seed;
   logger.sentry('Creating wallet, isImported?', isImported);
   const walletSeed = seed || generateSeedPhrase();
-  let addresses = [];
+  let addresses: RainbowAccount[] = [];
   try {
     const { hdnode, isHDWallet, type, wallet } = getWallet(walletSeed);
+    if (!wallet) return null;
     logger.sentry('[createWallet] - getWallet from seed');
 
     // Get all wallets
     const allWalletsResult = await getAllWallets();
     logger.sentry('[createWallet] - getAllWallets');
-    const allWallets = get(allWalletsResult, 'wallets', {});
+    const allWallets: AllRainbowWallets = get(
+      allWalletsResult,
+      'allWallets',
+      {}
+    );
 
     let existingWalletId = null;
     if (isImported) {
       // Checking if the generated account already exists and is visible
       logger.sentry('[createWallet] - isImported >> true');
-      const alreadyExistingWallet = find(allWallets, someWallet =>
-        find(
-          someWallet.addresses,
-          account =>
-            toChecksumAddress(account.address) ===
-              toChecksumAddress(wallet.address) && account.visible
-        )
-      );
+      let alreadyExistingWallet = null;
+      if (wallet) {
+        alreadyExistingWallet = find(
+          allWallets,
+          (someWallet: RainbowWallet) => {
+            return find(
+              someWallet.addresses,
+              account =>
+                toChecksumAddress(account.address) ===
+                  toChecksumAddress(wallet.address) && account.visible
+            )
+              ? true
```

Comment:
hm this ? true : false not needed. could do !!find if you'd like to be more explicit about returning a boolean value